pkg/platform/kube/logProxy/elastic/opensearch.go

```diff
@@ -0,0 +1,70 @@
+/*
+Copyright 2025 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package elastic
+
+import (
+	"context"
+	"crypto/tls"
+	"io"
+	"net/http"
+
+	"github.com/nuclio/nuclio/pkg/platform"
+	"github.com/nuclio/nuclio/pkg/platform/kube/logProxy"
+	"github.com/nuclio/nuclio/pkg/platformconfig"
+
+	"github.com/nuclio/errors"
+	"github.com/opensearch-project/opensearch-go/v4"
+)
+
+type OpenSearchLogProxy struct {
+	*AbstractElasticLikeLogProxy
+	client *opensearch.Client
+}
+
+func NewOpenSearchLogProxy(config *platformconfig.ElasticSearchConfig) (*OpenSearchLogProxy, error) {
+	openSearchClient := &OpenSearchLogProxy{
+		AbstractElasticLikeLogProxy: &AbstractElasticLikeLogProxy{
+			index:            config.Index,
+			customQueryParam: config.CustomQueryParameter,
+		},
+	}
+	var err error
+
+	tlsConfig := &tls.Config{
+		// Set to true to skip TLS verification if SSLVerificationMode is "none"
+		InsecureSkipVerify: config.SSLVerificationMode == "none",
+	}
```

Comment:
can also be in the abstract class

Reply:
@TomerShor Keep it as abstract class param even though it's required only for initialisation?

Reply:
in that case keep it like this.